monkey/monkey_island/cc/services/agent_plugin_service/flask_resources/agent_plugin_installation.py

```diff
@@ -0,0 +1,32 @@
+import logging
+from http import HTTPStatus
+
+from flask import make_response, request
+from flask_security import auth_token_required, roles_accepted
+
+from monkey_island.cc.flask_utils import AbstractResource
+from monkey_island.cc.services.authentication_service import AccountRole
+
+from .. import IAgentPluginService
+from ..errors import PluginInstallError
+
+logger = logging.getLogger(__name__)
+
+
+class AgentPluginInstallation(AbstractResource):
+    urls = ["/api/agent-plugins"]
```

Comment:
This URL already exists in the `AgentPlugins` resource, further supporting my point above.

Reply:
That url needs expects host_os, type and plugin name. If we did the PUT there and at thiss url would not that mean that we can PUT with those arguments and GET without those. I am not really into thr REST but that is what is confusing to me.

Reply:
Ilija is correct about why we did this. That being said, making it RPC does alleviate some issues. Making it an RPC-style `/api/install-agent-plugin` is a good idea.